botocore/retries/standard.py

```diff
@@ -198,7 +198,10 @@ def get_error_code(self):
         """
         if self.parsed_response is None:
             return
-        return self.parsed_response.get('Error', {}).get('Code')
+        error = self.parsed_response.get('Error', {})
```

Comment:
This seems sensible for the least amount of change. Do you know if there are currently services returning 200 responses for known error cases? I wonder if it would be better to just avoid calling this entirely for non-error status codes. I wouldn't expect us to retry on 100, 200, or 300 status codes.

Reply:
We discussed this offline and there may be some edge cases that make this a non-starter. We'll revisit avoiding this logic for status_code < 400 later.